### PR TITLE
fix(help): change help button to direct user to correct forum

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This expands the list of `challengeType`s that redirect to the JavaScript sub-forum.  Also, the 'Ask for help' button was not functioning for backend challenges - this has also been fixed.

In conjunction with freeCodeCamp/curriculum/pull/307 this closes issue #259

### CODE REVIEW

File: src/templates/Challenges/redux/create-question-epic.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ function createQuestionEpic(action$, { getState }, { window }) {
         '**Link to the challenge:**\n',
         href
       ].join('');
-      const categories = ['HTML-CSS', 'JavaScript'];
+      const categories = {0: 'HTML-CSS', 1: 'JavaScript', 2: 'JavaScript', 6: 'JavaScript'};
```
Comment (Bouncey) [inline src/templates/Challenges/redux/create-question-epic.js]: Is there a way we can cut down repartition here?

Maybe we could have a `helpCategory` export from `/utils/challengeTypes.js` which maps a challenge type to a category. This way we could manage all things relating to `challengeType` in one place.

File: src/templates/Challenges/backend/Show.js [hunk 2]
```diff
@@ -157,6 +158,7 @@ export class BackEnd extends PureComponent {
           <Spacer />
         </Col>
         <CompletionModal />
+        <HelpModal/>
```
Comment (Bouncey) [inline src/templates/Challenges/backend/Show.js]: Nice catch

File: src/templates/Challenges/redux/create-question-epic.js (hunk 1 above)
Comment (Bouncey) [reply src/templates/Challenges/redux/create-question-epic.js]: As an aside, the original category selection could have been extended by doing
```js
const categories = [ 'HTML-CSS', 'JavaScript',,,,,'Javascript' ];
```
But I am certain moving this to the `challengeTypes` file is a good idea

**Comment by Bouncey** [comment]
This PR will need to be updated with a new curriculum package version once the proposed curriculum fix is released

File: src/templates/Challenges/redux/create-question-epic.js (hunk 1 above)
Comment (ojeytonwilliams) [reply src/templates/Challenges/redux/create-question-epic.js]: I think this is a bit better.  Still a few repetitions, but it's in the right place now!

**Pull request merged**